.storybook/preview.ts

```diff
@@ -0,0 +1,147 @@
+    context: StoryContext,
+  ) => {
+    const selectedTheme = pluckThemeFromContext(context) || defaultTheme;
+    const themeCssUrl = themes[selectedTheme];
+
+    // Remove existing StoryBook theme CSS link if any
+    let existingLink = document.getElementById("storybook-theme-css");
+    if (existingLink) {
+      existingLink.parentNode?.removeChild(existingLink);
+    }
+
+    // Add new StoryBook theme CSS link
+    const link = document.createElement("link");
+    link.id = "storybook-theme-css";
+    link.rel = "stylesheet";
+    link.href = themeCssUrl;
+    document.head.appendChild(link);
+
+    return storyFn(context.args, context);
+  };
+};
+
+/**
+ * List of container sizes to cycle through
+ */
+const containerSizes = [
+  { title: "Full Width", value: "100%" },
+  { title: "Container (968px)", value: "968px" },
+  { title: "Container (856px)", value: "856px" },
+  { title: "Tablet (768px)", value: "768px" },
+  { title: "Container (668px)", value: "668px" },
+  { title: "Container (656px)", value: "656px" },
+  { title: "Container (543px)", value: "543px" },
+  { title: "Container (527px)", value: "527px" },
+  { title: "Container (476px)", value: "476px" },
+  { title: "Container (415px)", value: "415px" },
+  { title: "Mobile (320px)", value: "320px" },
+  { title: "Side Nav (288px)", value: "288px" },
+  { title: "Anchor Nav (168px)", value: "168px" },
+];
+
+/**
+ * Global Container Decorator
+ * Wraps all stories in multiple containers with adjustable sizes for Canvas.
+ * Displays only one example in the Docs page.
+ */
+const containerDecorator = (Story, context) => {
+  // Check if we are in the "docs" view mode
+  if (context.viewMode === "docs") {
+    // Show only one example (default size) in Docs
+    return html` ${Story()} `;
+  }
+
+  // In "story" view mode (Canvas), show all sizes
+  return html`
+    <div style="padding:1rem">
+      <div class="story-description-container">
+        <h2>Component Overview</h2>
+        <p>
+          This page demonstrates the
+          <strong>${context.name}</strong>
+          variant of the
+          <strong>${context.title}</strong>
+          component and its variations across the officially supported container
+          sizes. Use the controls to explore different configurations and
+          responsive behaviors.
+        </p>
+      </div>
+      ${containerSizes.map(
+        ({ title, value }) => html`
+          <div
+            style="
+              margin: 10px auto;
+              max-width: ${value};
+            "
+          >
+            <h4 style="text-align: center;">${title}</h4>
+            ${Story()}
+          </div>
+        `,
+      )}
+    </div>
+  `;
+};
+
+const preview: Preview = {
+  beforeEach({ canvasElement, canvas }) {
+    Object.assign(canvas, { ...withinShadow(canvasElement) });
+  },
+  decorators: [
+    // (Story, context) => containerDecorator(() => Story(context), context), // Add containerDecorator globally
```

Comment:
Borealis uses this decorator to render stories (only in the preview, not docs view) at a bunch of different viewport widths.

I found it rather unwieldy, but I've included it here if you prefer to use it too. Just uncomment this line.